### PR TITLE
Print where clauses

### PR DESCRIPTION
**Pull request by daviditen**
Update toString(FnSymbol*) to include the where clause if it exists

This will cause the where clause to be included in messages such as the
candidate list in an unresolved call error.

Updated a handful of test .good and .bad files to include where clauses.

This is requested in issue #5690.

### CODE REVIEW

**Comment by daviditen** [comment]
Passed a full linux64 paratest.

File: compiler/AST/symbol.cpp [hunk 1]
```diff
@@ -2216,10 +2211,16 @@ const char* toString(FnSymbol* fn) {
     if (skipParens == false) {
       retval = astr(retval, ")");
     }
+  }
 
-    if (developer  == true) {
-      retval = astr(retval, " [", istr(fn->id), "]");
-    }
+  if (fn->where && fn->where->body.length == 1) {
```
Comment (ben-albrecht) [inline compiler/AST/symbol.cpp]: Could you clarify the purpose of the `fn->where->body.length == 1` condition to me?

**Comment by ben-albrecht** [comment]
As a sanity check, have you tested any `where` clauses with more than 1 concatenated conditional? I didn't see any tests like that.

File: compiler/AST/symbol.cpp (hunk 1 above)
Comment (daviditen) [reply compiler/AST/symbol.cpp]: In the second commit message I mentioned it, but misspelled the word "shouldn't" as "should".

The compiler sometimes inserts multi-statement where clauses and we shouldn't print them.  One such case is for functions that have a queried argument.

**Comment by daviditen** [comment]
Ah, shoot.  You're right about the multiple conditions case.

```
proc foo(x) where isIntegral(x) && !isReal(x) { }
proc foo(x) where isReal(x) { }
foo("bar");
```

Doesn't print the where clause for the first version.

**Comment by daviditen** [comment]
I think I have a way make it work by caching the where clause earlier in the compiler before it gets normalized.  I'm going to close this PR for the moment though.  Thanks for looking at it.

**Comment by ben-albrecht** [comment]
I knew it was too good to be true!

**Pull request closed without merging**